src/operators/sample.ts

```diff
@@ -12,8 +12,8 @@ class SampleOperator<T, R> implements Operator<T, R> {
   constructor(private notifier: Observable<any>) {
   }
   
-  call(observer: Observer<R>) {
-    return new SampleSubscriber(observer, this.notifier);
+  call(subscriber: Subscriber<R>) {
```

Comment:
Not sure missing return type for some cases (sample, sampletime, etcs) are intended or not. To limit scope of changes, did not touch.